drivers/bus/ancillary_bus.c

```diff
@@ -0,0 +1,213 @@
+// SPDX-License-Identifier: GPL-2.0-only
+/*
+ * Software based bus for Ancillary devices
+ *
+ * Copyright (c) 2019-2020 Intel Corporation
+ *
+ * Please see Documentation/driver-api/ancillary_bus.rst for
+ * more information
+ */
+
+#include <linux/acpi.h>
+#include <linux/device.h>
+#include <linux/init.h>
+#include <linux/module.h>
+#include <linux/pm_domain.h>
+#include <linux/pm_runtime.h>
+#include <linux/string.h>
+#include <linux/ancillary_bus.h>
+
+static DEFINE_IDA(ancillary_dev_ida);
+#define ANCILLARY_INVALID_ID	~0U
+
+static const
+struct ancillary_device_id *ancillary_match_id(const struct ancillary_device_id *id,
+					   struct ancillary_device *adev)
+{
+	while (id->name[0]) {
+		if (!strcmp(adev->match_name, id->name))
+			return id;
+		id++;
+	}
+	return NULL;
+}
+
+static int ancillary_match(struct device *dev, struct device_driver *drv)
+{
+	struct ancillary_driver *adrv = to_ancillary_drv(drv);
+	struct ancillary_device *adev = to_ancillary_dev(dev);
+
+	return !!ancillary_match_id(adrv->id_table, adev);
```

Comment:
this function is only used once. and its small enough that it doesn't need to define as another static function.

Just put  the implementation here.

if you still want it as separate function, let it return bool, no need to return pointer id and use !!.
Keep code simple.

Reply:
changed